src/services/codefixes/inferFromUsage.ts

```diff
@@ -333,21 +333,50 @@ namespace ts.codefix {
     }
 
     function annotateJSDocParameters(changes: textChanges.ChangeTracker, sourceFile: SourceFile, parameterInferences: readonly ParameterInference[], program: Program, host: LanguageServiceHost): void {
-        const signature = parameterInferences.length && parameterInferences[0].declaration.parent;

```

Comment:
why not keep using `[0]` here? The code just checked that `parameterInferences.length > 0`, which is the only other thing that `first` does.